src/plugins/data/public/query/query_string/query_history.ts

```diff
@@ -14,12 +14,12 @@ export const HISTORY_KEY_PREFIX = 'query_';
 export class QueryHistory {
   private changeEmitter: BehaviorSubject<any[]>;
 
-  constructor(private readonly storage: DataStorage) {
+  constructor(private readonly sessionStorage: DataStorage) {
```

Comment:
is this variable change needed? i think `QueryHistory` isn't necessarily using sessionStorage, it depends on which storage is passed in when creating query history object

Reply:
Yeah I submit a follow up PR in this comment https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8723#discussion_r1819979917

Will merge this in first then the follow up one.